### ISSUE TITLE
callback for core solve progress 

### ISSUE THREAD

**Issue by pjcpjc** [ampl/amplpy#11]
Looks like the easiest way to do this is via #9, just need to worry about parsing the variable names, particularly if the entities have commas in them or are unicode. Might be a good opportunity for`ticdat` so swoop in with obfusimplification and some utility parsing routines. 

@fdabrandao seemed to think funky entity names won't be a problem, I think they will, but we can explore this. 

OTOH, if there is another way to implement "progress tracking and solve ending based on user feedback" that doesn't involve driving through `gurobipy`, I'm all ears.

**Comment by fdabrandao** [comment]
Regarding "progress tracking and solve ending based on user feedback", it may be worth to take a look at https://github.com/ampl/amplpy/blob/master/examples/asyncexample.py. I have now noticed that `ampl.interrupt()` does not seem to be working well, but would something like this be useful?

**Comment by pjcpjc** [comment]
Yes that is the better way to handle this, as opposed to back door-ing through `gurobipy`. 

We should work with this method first.

Do you have a slow MIP example - where a feasible solution is found fast, but the MIP gap closes slowly? If not I can try and make one. That is the use case we are targeting. MIPs that close a gap slowly, and the user decides that 1.5% is good enough after watching the UB/LB slowly creep towards each other for an hour.
